pages/_app.js

```diff
@@ -7,10 +7,32 @@ import { nanoid } from "nanoid";
 export default function App({ Component, pageProps }) {
   const [shoppingItems, setShoppingItems] = useState(initialShoppingItems);
 
+  const shoppingItemsWithCategoryColor = shoppingItems.map((shoppingItem) => {
+    const category = categories.find(
+      (category) => category.name === shoppingItem.category
+    );
+
+    const backgroundColor = category ? category.color : "white";
+
+    return {
+      ...shoppingItem,
+      backgroundColor,
+    };
+  });
+
+  const purchasedItems = shoppingItemsWithCategoryColor.filter(
+    (shoppingItem) => shoppingItem.isPurchased
+  );
+
+  const unpurchasedItems = shoppingItemsWithCategoryColor.filter(
+    (shoppingItem) => shoppingItem.isPurchased === false
+  );
+
   function handleAddItem(newItem) {
     setShoppingItems([
```

Comment:
I have replaced prevItems with shoppingItems.

I now call setShoppingItems, handleEditItem and handleDeleteItem with callback (prevItems) rather than directly as before.